packages/node/src/parseForPolicy.js

```diff
@@ -211,24 +264,39 @@ function makeImportHook({
     })
   }
 
+  /**
+   * Creates a CommonJS module record.
+   *
+   * @param {string} specifier - The module specifier.
+   * @param {string} filename - The filename of the module.
+   * @param {string} packageName - The name of the package.
+   * @returns {Promise<LavamoatModuleRecord>} A promise that resolves to a LavamoatModuleRecord object.
+   */
   async function makeJsModuleRecord(specifier, filename, packageName) {
     // load src
     const content = await fs.readFile(filename, 'utf8')
     // parse
     const ast = parseModule(content, specifier)
     // get imports
+    const { esmImports } = inspectEsmImports(ast, null, false)
     const { cjsImports } = inspectImports(ast, null, false)
+
+    // FIXME: if the package is CJS, and the extension is not `.mjs`, then the
+    // file cannot contain `import` statements. this will require reading the
+    // local `package.json` to determine, however.
+    const imports = [...new Set([...esmImports, ...cjsImports])]
```

Comment:
I think we end up reading every `package.json` one way or another.  So it'd be cool to have perhaps a LRU cache of them that we could pull from.  Such an API, if used here, would need to accept an arbitrary file and return its closest ancestor `package.json` (which may or may not be a cache hit).  

That is--_if_ we care to do anything about this.  A CJS script containing an `import` statement will fail with a `SyntaxError` anyway.

Reply:
AFAIR none of our tools (endo, lavamoat) polyfill the createRequire to be controlled by us. we could do that in the import hook we pass to importLocation